pkg/persistence/postgres/event.go

```diff
@@ -193,30 +204,92 @@ func (c *Event) EventLogDataToDB(payload *types.Log) {
 }
 
 // DBToEventLogData converts the DB raw log payload back to types.Log
-// NOTE(IS): because jsonb payloads are stored in DB as a map[string]interface{}, Postgres converts some fields, see notes in function.
+// NOTE(IS): because jsonb payloads are stored in DB as a map[string]interface{},
+// Postgres converts some fields, see notes in function.
 func (c *Event) DBToEventLogData() *types.Log {
-	log := &types.Log{}
-	log.Address = common.HexToAddress(c.LogPayload["Address"].(string))
-
-	topics := c.LogPayload["Topics"].([]interface{})
-	newTopics := make([]common.Hash, len(topics))
-	for i, topic := range topics {
-		topicString := topic.(string)
-		newTopics[i] = common.HexToHash(topicString)
+	tlog := &types.Log{}
+	tlog.Address = common.HexToAddress(c.LogPayload["Address"].(string))
+
+	tlog.Topics = c.typeInferTopics(c.LogPayload["Topics"])
+	tlog.Data = c.typeInferData(c.LogPayload["Data"])
+
+	// BlockNumber is stored in DB as float64
+	tlog.BlockNumber = c.typeInferBlockNumber(c.LogPayload["BlockNumber"])
```

Comment:
I don't think we should be inferring the type of what's being stored in the DB. We should know what it stores as and convert from that. What were the errors you got?

Reply:
The error that i was receiving the wrong type as interface (uint64 is not float, etc.) and it would panic out.  Could be that there are tests that are incorrect, but it seems worth it to me to cover for the these types to prevent the system from panic-ing out.

Reply:
Hm... I think we should then look in to the problem and fix those tests instead of adding this here? If you want to use this as a way to catch the problem. it would make more sense to me to only have 2 cases. One with the type we think it is, and the default to log which other types we are getting.

Reply:
Ok, let's do that then, I can work on that later today.